jupyterhub/handlers/login.py

```diff
@@ -82,6 +82,9 @@ async def get(self):
         """Log the user out, call the custom action, forward the user
         to the logout page
         """
+        if self.current_user is None:
+            self.log.debug("Got no user after redirect from code-server: redirected to logout url.")
+            self.redirect(self.settings['logout_url'], permanent=True)
```

Comment:
Unless there is a particular reason, I would set `permanent` to false to avoid any caching behavior of browsers